### ISSUE TITLE
Add tax form support for cross-host expenses

### ISSUE THREAD

**Issue by Betree** [opencollective/opencollective#5068]
Issue reported in https://opencollective.freshdesk.com/a/tickets/51876
Part of https://github.com/opencollective/opencollective/issues/5064

When looking for tax form statuses, we group entries by `Expense.FromCollectiveId`. With the introduction of tax forms across hosts, the payee can be a hosted initiative if the collective has an active fiscal host (`fromCollective.HostCollectiveId`) we should use it instead. We should also think about the UX and wording, as the person who needs to fill in this tax form is one of the host admins - not the collective admin.

Current workaround: manually mark tax form as received in the database. This is not ideal, and we should revert these entries once the fix will be live. Also, we should manually mark tax forms as received for the hosts we control (OSC, OCF) which should already resolve many situations.

**Comment by alanna** [comment]
> We should manually mark tax forms as received for the hosts we control (OSC, OCF) which should already resolve many situations.

Has this part been done @Betree ? 

Season of Docs 2022 will launch soon and will involve expenses across non-1st party hosts so we probably need to get this sorted.

**Comment by Betree** [comment]
> > We should manually mark tax forms as received for the hosts we control (OSC, OCF) which should already resolve many situations.
> 
> Has this part been done @Betree ?
> 
> Season of Docs 2022 will launch soon and will involve expenses across non-1st party hosts so we probably need to get this sorted.

Yes, tax forms have been marked as received for OSC (in 2021) and OCF (in 2020). Tax forms now being valid for 3 years, we should be goood.

**Comment by alanna** [comment]
When a Collective not under OCF (an IC or under another host like NumFOCUS) submits an expense to Season of Docs under OSC, how will this be handled? 

For an IC, it seems straightforward: the tax form notification should go to the admins and OSC needs to collect a tax form in the name of the legal entity being paid. However, we end up with a lot of 'hosts' who are actually supposed to be ICs but have set themselves up as a host with no Collectives instead. So we can't rely on the IC distinction fully.

For a host, such a NumFOCUS, we need to think it through, because the expense should be treated as coming from the Collective in terms of where the money is going, but the tax form needs to come from the host, and the notification should be sent to the _host_ admin. It's also important that the host is only prompted to submit it once, even though multiple Collectives under their umbrella might submit expenses.

**Comment by Betree** [comment]
Got a similar case in https://opencollective.freshdesk.com/a/tickets/62872

Feedback from @alanna:
> The tax form in such a case is the HOST’s tax form. So it’s the host admins that need to be asked to fill it out. A Collective admin is not authorized to sign such a form on behalf of the host. If it’s an IC then we can safely assume the admin can provide it. If it’s a hosted Collective they cannot. So to generalise the flow, we should trigger the tax form to be signed by the host admin, and have the tax form received status applied to all that Host’s Collectives so it doesn’t become a duplicate request. We should go in and turn this requirement off between first party host’s though, because we can manage that internally (for expenses between OSC/OCF/OCEU). For third party hosts, OSC and OCF still needs their tax form so it should be treated like an organization submitting an expense.

**Comment by znarf** [comment]
Todo:

- [ ] modify `getTaxFormsRequiredForExpenses` to check
- [ ] modify `getTaxFormsRequiredForAccounts` to check
- [ ] modify text of tax form banner
- [ ] modify wording in email

**Comment by Betree** [comment]
I closed this bit quickly. API support is basically done but we still need to tackle the frontend/docs parts items from https://github.com/opencollective/opencollective/issues/5068#issuecomment-1073703106

**Comment by Betree** [comment]
Regarding `modify text of tax form banner`, we have multiple strings concerned by this:
- On the expense status tag: `We can't pay until we receive your tax info. Check your inbox for an email from HelloWorks. Need help? Contact support@opencollective.com`. This one doesn't need an update IMO.
- On the expense page, after submission:
> We need your tax information before we can pay you. You will receive an email with a link to fill out a form. If you have not received the email within 24 hours, check your spam, then contact <I18nSupportLink></I18nSupportLink>. Questions? See <Link>help docs about taxes</Link>.

I think we can implement the following logic there:

```es6
const getTaxFormMessage = (loggedInUser, expense) => {
  if (expense.payee.host) {
    if (loggedInUser.isAdmin(expense.payee.host)) {
      return `As ${expense.payee.name}'s host, we need your tax form before we can pay this expense. (...)`;
    } else if (loggedInUser.isAdmin(expense.payee)) {
      return `We need your fiscal host, ${expense.payee.host.name}, to submit a tax form before this expense can be paid. They should have received an email about that, but feel free to contact them to give them more info`;
    }
  } else {
    if (loggedInUser.isAdmin(expense.payee)) {
      return '...'; // Current message (id="expenseNeedsTaxFormMessage.msg")
    } else {
      return `We need ${expense.payee.name} to submit their tax information before this expense can be paid`;
    }
  }
}
```
